server/game/cards/characters/01/oldforesthunter.js

```diff
@@ -4,30 +4,16 @@ class OldForestHunter extends DrawCard {
     setupCardAbilities(ability) {
         this.action({
             title: 'Discard a card to gain 1 gold',
-            method: 'discard',
-            limit: ability.limit.perPhase(1)
+            condition: () => !this.controller.cannotGainGold,
+            cost: ability.costs.discardFromHand(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                this.game.addGold(this.controller, 1);
```

Comment:
this should probably be moved after the messagae below, as otherwise the log will see first the message about "cannot gain gold", and later then one about gaining gold, which would be confusing, I guess.

Off the top of my head, logging first and then doing the action is also the most common message logging patterns in card implementations (at least in recent card implementations, that is).